### PR TITLE
Add navPlace to ScannedMaps

### LINKED ISSUE #4732: Add navPlace extension to ScannedMap manifests
https://iiif.io/api/extension/navplace/

1. Generate Presentation v3 manifests and add new route for them
1. Generate navPlace extensions in v3 manifest

### PR DESCRIPTION
**Pull request by eliotjordan**
Closes #4732

### CODE REVIEW

File: app/services/manifest_builder_v3.rb [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+    def source_metadata_identifier
+      resource.try(:source_metadata_identifier)
+    end
+
+    ##
+    # @param [Resource] resource the Resource being modeled as the root
+    def initialize(resource, auth_token = nil, current_ability = nil)
+      @resource = resource
+      @auth_token = auth_token
+      @current_ability = current_ability
```
Comment (tpendragon) [inline app/services/manifest_builder_v3.rb:39]: Question: This seems to be the case now too, but even if I don't have permission to a child resource it'll get added to the manifest?

File: config/routes.rb [hunk 2]
```diff
@@ -74,6 +74,7 @@
   end
 
   get "/downloads/:resource_id/file/:id", to: "downloads#show", as: :download
+  get "/manifests/:id/v3", to: "manifests#v3", as: :manifest_v3, defaults: { format: :json }
```
Comment (tpendragon) [inline config/routes.rb:77]: Comment: We'll have to add this route to nginx_extra_config in princeton_ansible for CORS.

**Review by tpendragon — APPROVED** [review]
Just one question.

File: app/services/manifest_builder_v3.rb (hunk 1 above)
Comment (eliotjordan) [reply app/services/manifest_builder_v3.rb:39]: Yes, it's the case now. I can't imagine, though, a case where someone would have permission on the parent and not the child.

File: config/routes.rb (hunk 2 above)
Comment (eliotjordan) [reply config/routes.rb:77]: https://github.com/pulibrary/princeton_ansible/issues/2551

**Pull request merged**